internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Parser.scala

```diff
@@ -150,8 +150,13 @@ private[sbt] object Parser {
                 case '[' =>
                   for (_ <- 0 until in.readUnsignedShort())
                     parseElementValue()
+                case 'B' | 'C' | 'D' | 'F' | 'I' | 'J' | 'S' | 'Z' | 's' =>
+                  val _ = in.readUnsignedShort()
                 case _ =>
-                // ignore. robustness is paramount
+                  // what to do here? perhaps robustness should be paramount, but if we encounter
+                  // something unexpected here, we're probably already doomed and attempting to
+                  // continue parsing will just make troubleshooting harder. so let's bail
+                  sys.error(s"unexpected tag in annotation: '$c'")
```

Comment:
Could we pretend you didn't see the annotation? It's been sort of ok thus far ignoring the annotations, so sometimes ignoring it sound like a safe middle ground than blocking Java compilation even for clean compile.

Reply:
Yes, good idea, will do.